lib/structs/query/query_preset_details.py

```diff
@@ -0,0 +1,17 @@
+from dataclasses import dataclass
+from enum import Enum
+
+from enums.query.query_presets import QueryPresets
+from typing import Any, Dict
+
+
+@dataclass
+class QueryPresetDetails:
+    """
+    Structured data passed to a Query<Resource> object when calling the where() function
+    describes how to filter for the query.
+    """
+
+    preset: QueryPresets
+    prop: Enum
```

Comment:
Could we specify which `Enum` or add a stub base class which such as `PropEnum` to subset this down?